modules/account-lib/src/utils/crypto.ts

```diff
@@ -103,3 +105,48 @@ export function isValidPrv(prv: string): boolean {
   }
   return true;
 }
+
+/**
+ * Whether the input is a valid ed25519 private key
+ *
+ * @param {string} prv A hexadecimal private key to validate
+ * @returns {boolean} Whether the input is a valid public key or not
+ */
+export function isValidEd25519Seed(prv: string): boolean {
+  try {
+    const decodedPrv = toUint8Array(prv);
+    return decodedPrv.length === nacl.sign.seedLength;
+  } catch (e) {
+    return false;
+  }
+}
+
+/**
+ * Whether the input is a valid ed25519 private key
+ *
+ * @param {string} prv A hexadecimal private key to validate
+ * @returns {boolean} Whether the input is a valid public key or not
+ */
+export function isValidEd25519SecretKey(prv: string): boolean {
+  try {
+    const decodedPrv = toUint8Array(prv);
+    return decodedPrv.length === nacl.sign.secretKeyLength;
```

Comment:
basically same comment as above

Reply:
also is it possible that the seed for any derived coin is much larger than this length?

Reply:
so maybe this is where it gets weird. i think we should use like an encoded format to ensure this is one of ours. a prefix maybe? 

I'll leave this to chew on later for now